lib/D2L.Extensibility.AuthSdk.UnitTests/FunctionalApiUriTests.cs

```diff
@@ -288,6 +288,32 @@ public class FunctionalApiUriTests {
 		}
 
 		[Test]
+		public void UserContext_CreateAuthUri_QuerystringIsMaintained() {
+			Uri queryUrl = m_userContext.CreateAuthenticatedUri( TestConstants.API_PATH_WITH_QUERY, "GET" );
+
+			Assert.IsTrue(
+				queryUrl.ToString().Contains( TestConstants.API_PATH_QUERYSTRING ),
+				"The querystring parameter was not present"
+				);
+		}
+
+		[Test]
+		public void UserContext_CreateAuthUri_MultipleQuerystringsAreNotPresent() {
+			const string querystringCharacterEncoded = "%3F";
+
+			Uri queryUrl = m_userContext.CreateAuthenticatedUri( TestConstants.API_PATH_WITH_QUERY, "GET" );
+			bool duplicatedQuery = (

```

Comment:
Don't think it's bad. Just feel like if we can provide a test that says and
verifies what SHOULD happen, instead of many tests that specify what
shouldn't - we're better off (in this case the should would cover the
should not).

I mean, these tests are sufficient and we can surge ahead confidently. I
think I will likely add in the test I mentioned though, as it would help me
as a developer in creating a new implementation for instance.

On Tue, May 12, 2015 at 12:55 PM, Kyle Lewis <notifications@github.com>
wrote:

> In lib/D2L.Extensibility.AuthSdk.UnitTests/FunctionalApiUriTests.cs
> <https://github.com/Brightspace/valence-sdk-dotnet/pull/5#discussion_r30155038>
> :
>
> > @@ -288,6 +288,32 @@ public class FunctionalApiUriTests {
> >  		}
> >
> >  		[Test]
> > +		public void UserContext_CreateAuthUri_QuerystringIsMaintained() {
> > +			Uri queryUrl = m_userContext.CreateAuthenticatedUri( TestConstants.API_PATH_WITH_QUERY, "GET" );
> > +
> > +			Assert.IsTrue(
> > +				queryUrl.ToString().Contains( TestConstants.API_PATH_QUERYSTRING ),
> > +				"The querystring parameter was not present"
> > +				);
> > +		}
> > +
> > +		[Test]
> > +		public void UserContext_CreateAuthUri_MultipleQuerystringsAreNotPresent() {
> > +			const string querystringCharacterEncoded = "%3F";
> > +
> > +			Uri queryUrl = m_userContext.CreateAuthenticatedUri( TestConstants.API_PATH_WITH_QUERY, "GET" );
> > +			bool duplicatedQuery = (
>
> Hmm... I had added the test in question as specific regression test. Prior
> to the change to D2LUserContext, both the "?" and "%3F" separator were
> present in the same URL.
>
> In your opinion, is this a bad test because it's checking for something
> specific or because it's not clear on what it's testing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Brightspace/valence-sdk-dotnet/pull/5/files#r30155038>
> .
>



-- 
Owen